logger.go

```diff
@@ -96,6 +96,13 @@ func (logger *Logger) Logf(level LogLevel, format string, args ...interface{}) {
 	}
 }
 
+func (logger *Logger) Write(p []byte) (n int, err error) {
+	logEntryMessage := NewLogEntryMessage(logger.scopes, InfoLevel, string(p))
```

Comment:
This looks like a [format string injection](https://capec.mitre.org/data/definitions/135.html). Even through this is not C, I think it might be a more sound approach to pass an already formatted message to the `NewLogEntryMessage()` and simplify `LogEntryMessage` struct as a result.